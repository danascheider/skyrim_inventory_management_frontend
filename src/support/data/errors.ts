export const authorizationErrorResponse = {
  errors: ['401 Unauthorized'],
}

export const notFoundResponse = {
  errors: ['404 Not Found'],
}

export const internalServerErrorResponse = {
  errors: ['500 Internal Server Error'],
}
