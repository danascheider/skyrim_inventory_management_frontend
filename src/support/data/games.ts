import { ResponseGame as Game } from '../../types/apiData'

export const emptyGames: Game[] = []

export const allGames: Game[] = [
  {
    id: 32,
    user_id: 412,
    name: 'My Game 1',
    description: 'This is a game with a description',
    created_at: new Date(),
    updated_at: new Date(),
  },
  {
    id: 51,
    user_id: 412,
    name: 'My Game 2',
    description: null,
    created_at: new Date(),
    updated_at: new Date(),
  },
  {
    id: 77,
    user_id: 412,
    name: 'Game with a really really really really really long name',
    description:
      'Cum audissem Antiochum, Brute, ut solebam, cum M. Pisone in eo gymnasio, quod Ptolomaeum vocatur, unaque nobiscum Q. frater et T. Pomponius postmeridianam conficeremus in Academia',
    created_at: new Date(),
    updated_at: new Date(),
  },
]
