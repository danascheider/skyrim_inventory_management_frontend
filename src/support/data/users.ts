import { User } from 'firebase/auth'
import testProfileImg from '../testProfileImg.png'

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdtoken: () =>
    new Promise<string>((resolve, _reject) => resolve('xxxxxxx')),
} as unknown as User
