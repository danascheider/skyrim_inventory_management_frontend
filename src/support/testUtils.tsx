import { type ReactElement } from 'react'
import { BrowserRouter } from 'react-router-dom'
import { JSDOM } from 'jsdom'
import { render } from '@testing-library/react'
import { LoginContext } from '../contexts/loginContext'
import { User } from 'firebase/auth'
import testProfileImg from './testProfileImg.png'

export const BASE_APP_URI = 'http://localhost:5173'

declare global {
  namespace NodeJS {
    interface Global {
      document: Document
      window: Window
    }
  }
}

const setDom = () => {
  const dom = new JSDOM('<!doctype html><html><body></body></html>', {
    url: 'http://localhost:5173',
  })

  global.window = dom.window as unknown as Window & typeof globalThis
  global.document = dom.window.document
}

/*
 *
 * Global Test Data
 *
 */

export const requireLogin = () => {
  /* noop */
}

export const testUser = {
  uid: 'edna',
  displayName: 'Edna St. Vincent Millay',
  email: 'edna@gmail.com',
  photoURL: testProfileImg,
  emailVerified: true,
  getIdtoken: () =>
    new Promise<string>((resolve, _reject) => resolve('xxxxxxx')),
} as unknown as User

/*
 *
 * Test Renderers
 *
 */

const renderWithJSDOM = (ui: ReactElement) => {
  setDom()

  return render(ui)
}

export const renderWithRouter = (ui: ReactElement) =>
  renderWithJSDOM(<BrowserRouter>{ui}</BrowserRouter>)

export const renderAuthenticated = (ui: ReactElement, authLoading = false) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: testUser,
        token: 'xxxxxxx',
        requireLogin,
        authLoading,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderUnauthenticated = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{
        user: null,
        token: null,
        authLoading: false,
        requireLogin,
      }}
    >
      {ui}
    </LoginContext.Provider>
  )

export const renderAuthLoading = (ui: ReactElement) =>
  renderWithRouter(
    <LoginContext.Provider
      value={{ user: null, token: null, authLoading: true, requireLogin }}
    >
      {ui}
    </LoginContext.Provider>
  )
