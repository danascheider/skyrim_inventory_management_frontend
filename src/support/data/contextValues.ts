import { type GamesContextType } from '../../contexts/gamesContext'
import { type LoginContextType } from '../../contexts/loginContext'
import { type ShoppingListsContextType } from '../../contexts/shoppingListsContext'
import { DONE, ERROR, LOADING } from '../../utils/loadingStates'
import { testUser } from './users'
import { allGames } from './games'
import { emptyShoppingLists, shoppingListsForGame } from './shoppingLists'

const noop = () => {}

/**
 *
 * Default values for Login context
 *
 */

export const loginContextValue: LoginContextType = {
  user: testUser,
  token: 'xxxxxxx',
  requireLogin: noop,
  authLoading: false,
}

export const loadingLoginContextValue: LoginContextType = {
  user: null,
  token: null,
  requireLogin: noop,
  authLoading: true,
}

export const unauthenticatedLoginContextValue: LoginContextType = {
  user: null,
  token: null,
  requireLogin: noop,
  authLoading: false,
}

/**
 *
 * Default value for Games context
 *
 */

export const gamesContextValue: GamesContextType = {
  games: allGames,
  gamesLoadingState: DONE,
  createGame: noop,
  updateGame: noop,
  destroyGame: noop,
}

export const gamesContextValueLoading: GamesContextType = {
  games: [],
  gamesLoadingState: LOADING,
  createGame: noop,
  updateGame: noop,
  destroyGame: noop,
}

export const gamesContextValueError: GamesContextType = {
  games: [],
  gamesLoadingState: ERROR,
  createGame: noop,
  updateGame: noop,
  destroyGame: noop,
}

/**
 *
 * Default values for Shopping Lists context
 *
 */

export const shoppingListsContextValueEmpty: ShoppingListsContextType = {
  shoppingLists: emptyShoppingLists,
  shoppingListsLoadingState: DONE,
  createShoppingList: noop,
}

export const shoppingListsContextValue: ShoppingListsContextType = {
  shoppingLists: shoppingListsForGame(77),
  shoppingListsLoadingState: DONE,
  createShoppingList: noop,
}

export const shoppingListsContextValueLoading: ShoppingListsContextType = {
  shoppingLists: emptyShoppingLists,
  shoppingListsLoadingState: LOADING,
  createShoppingList: noop,
}

export const shoppingListsContextValueError: ShoppingListsContextType = {
  shoppingLists: emptyShoppingLists,
  shoppingListsLoadingState: ERROR,
  createShoppingList: noop,
}
