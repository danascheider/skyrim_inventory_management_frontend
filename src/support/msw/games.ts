import { rest } from 'msw'
import { ResponseGame } from '../../types/apiData'
import { emptyGames, allGames } from '../data/games'

const BASE_URI = 'http://localhost:3000'

/**
 *
 * POST /games
 *
 */

export const postGamesSuccess = rest.post(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    const body = {
      id: 102,
      user_id: 412,
      name: req.params.name || 'My Game 3',
      description:
        req.params.description || 'This description is just for illustration.',
      created_at: new Date(),
      updated_at: new Date(),
    }

    return res(ctx.status(201), ctx.json(body))
  }
)

export const postGamesUnprocessable = rest.post(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    const body = {
      errors: [
        "Name can only contain alphanumeric characters, spaces, commas (,), hyphens (-), and apostrophes (')",
      ],
    }

    return res(ctx.status(422), ctx.json(body))
  }
)

export const postGamesServerError = rest.post(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    const body = { errors: ['oh noes'] }

    return res(ctx.status(500), ctx.json(body))
  }
)

/**
 *
 * GET /games
 *
 */

export const getGamesEmptySuccess = rest.get(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyGames))
  }
)

export const getGamesAllSuccess = rest.get(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(allGames))
  }
)

export const getGamesServerError = rest.get(
  `${BASE_URI}/games`,
  (req, res, ctx) => {
    return res(ctx.status(500), ctx.json({ errors: ['Something went wrong'] }))
  }
)

/**
 *
 * PATCH /games/:id
 *
 */

const newOrExistingGame = (id: number): ResponseGame => {
  const existingGame = allGames.find((game) => game.id === id)

  if (existingGame) return existingGame

  return {
    id,
    user_id: 412,
    name: 'Skyrim Game',
    description: null,
    created_at: new Date(),
    updated_at: new Date()
  }
}

export const patchGameSuccess = rest.patch(
  `${BASE_URI}/games/:id`,
  async (req, res, ctx) => {
    const id = Number(req.params.id)
    const body = await req.json()
    const game = newOrExistingGame(id)

    return res(
      ctx.status(200),
      ctx.json({ ...game, ...body })
    )
  }
)

export const patchGameUnprocessableEntity = rest.patch(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
  return res(
    ctx.status(422),
    ctx.json({ errors: ['Name must be unique'] })
  )
})

export const patchGameNotFound = rest.patch(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
    return res(
      ctx.status(404)
    )
  }
)

export const patchGameServerError = rest.patch(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
    return res(
      ctx.status(500),
      ctx.json({ errors: ['oh noes'] })
    )
  }
)

/**
 *
 * DELETE /games/:id
 *
 */

export const deleteGameSuccess = rest.delete(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
    return res(ctx.status(204))
  }
)

export const deleteGameNotFound = rest.delete(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
    return res(ctx.status(404))
  }
)

export const deleteGameServerError = rest.delete(
  `${BASE_URI}/games/:id`,
  (req, res, ctx) => {
    return res(ctx.status(500), ctx.json({ errors: ['oh noes'] }))
  }
)
