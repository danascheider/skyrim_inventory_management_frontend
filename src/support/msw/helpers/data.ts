import {
  type RequestShoppingListItem,
  type RequestShoppingList,
  type ResponseShoppingListItem,
  type ResponseShoppingList,
} from '../../../types/apiData'
import { allGames } from '../../data/games'
import {
  allShoppingLists,
  shoppingListsForGame,
} from '../../data/shoppingLists'

const gameIds = allGames.map(({ id }) => id)

/**
 *
 * Shopping List Creation
 *
 */

export const newShoppingList = (
  attributes: RequestShoppingList,
  gameId: number
): ResponseShoppingList[] => {
  if (gameIds.indexOf(gameId) < 0)
    throw new Error(
      'Cannot generate shopping list for game that does not exist in test data'
    )

  const existingLists = shoppingListsForGame(gameId)

  if (!existingLists.length)
    throw new Error(
      'Cannot generate single list for game without existing aggregate'
    )

  return [
    {
      id: 93,
      game_id: gameId,
      aggregate: false,
      aggregate_list_id: existingLists[0].id,
      title: attributes.title || 'New Shopping List',
      list_items: [],
      created_at: new Date(),
      updated_at: new Date(),
    },
  ]
}

export const newShoppingListWithAggregate = (
  attributes: RequestShoppingList,
  gameId: number
): ResponseShoppingList[] => {
  if (gameIds.indexOf(gameId) < 0)
    throw new Error(
      'Cannot generate shopping list for game that does not exist in test data'
    )

  const existingLists = shoppingListsForGame(gameId)

  if (existingLists.length)
    throw new Error(
      'Cannot generate new aggregate list for game that already has one'
    )

  return [
    {
      id: 93,
      game_id: gameId,
      aggregate: true,
      aggregate_list_id: null,
      title: 'All Items',
      list_items: [],
      created_at: new Date(),
      updated_at: new Date(),
    },
    {
      id: 94,
      game_id: gameId,
      aggregate: false,
      aggregate_list_id: 93,
      title: attributes.title || 'My Shopping List 1',
      list_items: [],
      created_at: new Date(),
      updated_at: new Date(),
    },
  ]
}

/**
 *
 * Shopping list item creation
 *
 */

export const newShoppingListItem = (
  attributes: RequestShoppingListItem,
  listId: number
) => {
  const list = allShoppingLists.find(({ id }) => id === listId)

  if (!list)
    throw new Error(`No shopping list with ID ${listId} in the test data`)

  const shoppingList = { ...list }
  const allLists = shoppingListsForGame(shoppingList.game_id)
  const aggregateList = { ...allLists[0] }

  const newItem: ResponseShoppingListItem = {
    id: 42,
    list_id: listId,
    description: 'Dummy description for TypeScript',
    quantity: 1,
    unit_weight: null,
    notes: null,
    created_at: new Date(),
    updated_at: new Date(),
    ...attributes,
  }

  const newAggregateListItem: ResponseShoppingListItem = {
    ...newItem,
    id: 43,
    list_id: aggregateList.id,
  }

  aggregateList.list_items = [newAggregateListItem, ...aggregateList.list_items]
  shoppingList.list_items = [newItem, ...shoppingList.list_items]

  return [aggregateList, shoppingList]
}
