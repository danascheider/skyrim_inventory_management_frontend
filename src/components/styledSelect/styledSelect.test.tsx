import { describe, test, expect } from 'vitest'
import { act, fireEvent } from '@testing-library/react'
import { render } from '../../support/testUtils'
import { allGames } from '../../support/data/games'
import StyledSelect from './styledSelect'

/**
 *
 * Not able to be tested: disabled state
 *
 */

describe('StyledSelect', () => {
  describe('when there are options', () => {
    const options = allGames.map(({ name, id }) => ({
      optionName: name,
      optionValue: id,
    }))

    test('renders the options', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      // Doesn't display placeholder text since options exist
      expect(wrapper.queryByText("Doesn't matter")).toBeFalsy()

      // Main, initially visible box is empty by default
      expect(wrapper.getByTestId('selectedOption').textContent).toBeFalsy()

      // Initially, no option is selected
      expect(wrapper.queryByRole('option', { selected: true })).toBeFalsy()

      // All the option names should be present in the DOM
      expect(wrapper.getByText('My Game 1')).toBeTruthy()
      expect(wrapper.getByText('My Game 2')).toBeTruthy()
      expect(wrapper.getByText('Game with a really real...')).toBeTruthy()
    })

    test('matches snapshot', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when there are no options', () => {
    test('displays the placeholder text', () => {
      const wrapper = render(
        <StyledSelect
          options={[]}
          onOptionSelected={() => {}}
          placeholder="No options available"
        />
      )

      expect(wrapper.getByText('No options available')).toBeTruthy()
    })

    test('truncates the placeholder text if it is too long', () => {
      const wrapper = render(
        <StyledSelect
          options={[]}
          onOptionSelected={() => {}}
          placeholder="This placeholder is way too long."
        />
      )

      expect(wrapper.getByTestId('selectedOption').textContent).toEqual(
        'This placeholder is way...'
      )
    })

    test('matches snapshot', () => {
      const wrapper = render(
        <StyledSelect
          options={[]}
          onOptionSelected={() => {}}
          placeholder="No options available"
        />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when a default option is given', () => {
    const options = allGames.map(({ name, id }) => ({
      optionName: name,
      optionValue: id,
    }))

    test('selects the option and displays the selected option text', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          defaultOption={options[1]}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      // The placeholder text shouldn't be shown
      expect(wrapper.queryByText("Doesn't matter")).toBeFalsy()

      // The default option should have aria-selected set to "true"
      expect(
        wrapper.getByRole('option', { selected: true }).textContent
      ).toEqual(options[1].optionName)

      // The header text should be the option name
      expect(wrapper.getByTestId('selectedOption').textContent).toEqual(
        options[1].optionName
      )
    })

    test('truncates the title if it is too long', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          defaultOption={options[2]}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      // The header text should be the option name
      expect(wrapper.getByTestId('selectedOption').textContent).toEqual(
        'Game with a really real...'
      )
    })

    test('matches snapshot', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          defaultOption={options[1]}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('selecting an option', () => {
    const options = allGames.map(({ name, id }) => ({
      optionName: name,
      optionValue: id,
    }))

    test('sets the aria-selected attribute', () => {
      const wrapper = render(
        <StyledSelect
          options={options}
          defaultOption={options[1]}
          onOptionSelected={() => {}}
          placeholder="Doesn't matter"
        />
      )

      const option = wrapper.getByText(options[0].optionName)

      act(() => fireEvent.click(option))

      expect(
        wrapper.getByRole('option', { selected: true }).textContent
      ).toEqual(options[0].optionName)
    })
  })
})
