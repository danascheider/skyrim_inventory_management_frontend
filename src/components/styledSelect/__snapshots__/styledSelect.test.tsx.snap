// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyledSelect > when a default option is given > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_0a8ecf"
        data-testid="styledSelect"
        style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
      >
        <div
          aria-controls="selectListbox"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="selectListbox"
          class="_header_0a8ecf"
          role="combobox"
        >
          <p
            class="_headerText_0a8ecf"
            data-testid="selectedOption"
          >
            My Game 2
          </p>
          <button
            class="_trigger_0a8ecf"
            data-testid="selectTrigger"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down _fa_0a8ecf"
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <ul
          class="_dropdown_0a8ecf _hidden_0a8ecf"
          id="selectListbox"
          role="listbox"
        >
          <li
            aria-selected="false"
            class="_root_b75fe4"
            data-option-value="32"
            role="option"
            tabindex="0"
          >
            My Game 1
          </li>
          <li
            aria-selected="true"
            class="_root_b75fe4"
            data-option-value="51"
            role="option"
            tabindex="0"
          >
            My Game 2
          </li>
          <li
            aria-selected="false"
            class="_root_b75fe4"
            data-option-value="77"
            role="option"
            tabindex="0"
          >
            Game with a really real...
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_0a8ecf"
      data-testid="styledSelect"
      style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
    >
      <div
        aria-controls="selectListbox"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="selectListbox"
        class="_header_0a8ecf"
        role="combobox"
      >
        <p
          class="_headerText_0a8ecf"
          data-testid="selectedOption"
        >
          My Game 2
        </p>
        <button
          class="_trigger_0a8ecf"
          data-testid="selectTrigger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-down _fa_0a8ecf"
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <ul
        class="_dropdown_0a8ecf _hidden_0a8ecf"
        id="selectListbox"
        role="listbox"
      >
        <li
          aria-selected="false"
          class="_root_b75fe4"
          data-option-value="32"
          role="option"
          tabindex="0"
        >
          My Game 1
        </li>
        <li
          aria-selected="true"
          class="_root_b75fe4"
          data-option-value="51"
          role="option"
          tabindex="0"
        >
          My Game 2
        </li>
        <li
          aria-selected="false"
          class="_root_b75fe4"
          data-option-value="77"
          role="option"
          tabindex="0"
        >
          Game with a really real...
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StyledSelect > when there are no options > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_0a8ecf"
        data-testid="styledSelect"
        style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
      >
        <div
          aria-controls="selectListbox"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="selectListbox"
          class="_header_0a8ecf"
          role="combobox"
        >
          <p
            class="_headerText_0a8ecf"
            data-testid="selectedOption"
          >
            No options available
          </p>
          <button
            class="_trigger_0a8ecf"
            data-testid="selectTrigger"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down _fa_0a8ecf"
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <ul
          class="_dropdown_0a8ecf _hidden_0a8ecf"
          id="selectListbox"
          role="listbox"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_0a8ecf"
      data-testid="styledSelect"
      style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
    >
      <div
        aria-controls="selectListbox"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="selectListbox"
        class="_header_0a8ecf"
        role="combobox"
      >
        <p
          class="_headerText_0a8ecf"
          data-testid="selectedOption"
        >
          No options available
        </p>
        <button
          class="_trigger_0a8ecf"
          data-testid="selectTrigger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-down _fa_0a8ecf"
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <ul
        class="_dropdown_0a8ecf _hidden_0a8ecf"
        id="selectListbox"
        role="listbox"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StyledSelect > when there are options > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_0a8ecf"
        data-testid="styledSelect"
        style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
      >
        <div
          aria-controls="selectListbox"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="selectListbox"
          class="_header_0a8ecf"
          role="combobox"
        >
          <p
            class="_headerText_0a8ecf"
            data-testid="selectedOption"
          />
          <button
            class="_trigger_0a8ecf"
            data-testid="selectTrigger"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-down _fa_0a8ecf"
              data-icon="angle-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <ul
          class="_dropdown_0a8ecf _hidden_0a8ecf"
          id="selectListbox"
          role="listbox"
        >
          <li
            aria-selected="false"
            class="_root_b75fe4"
            data-option-value="32"
            role="option"
            tabindex="0"
          >
            My Game 1
          </li>
          <li
            aria-selected="false"
            class="_root_b75fe4"
            data-option-value="51"
            role="option"
            tabindex="0"
          >
            My Game 2
          </li>
          <li
            aria-selected="false"
            class="_root_b75fe4"
            data-option-value="77"
            role="option"
            tabindex="0"
          >
            Game with a really real...
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_0a8ecf"
      data-testid="styledSelect"
      style="--button-background-color: #2274a5; --button-hover-color: #1e6894; --button-text-color: #fff; --button-border-color: #1b5c84;"
    >
      <div
        aria-controls="selectListbox"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="selectListbox"
        class="_header_0a8ecf"
        role="combobox"
      >
        <p
          class="_headerText_0a8ecf"
          data-testid="selectedOption"
        />
        <button
          class="_trigger_0a8ecf"
          data-testid="selectTrigger"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-down _fa_0a8ecf"
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M169.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 274.7 54.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <ul
        class="_dropdown_0a8ecf _hidden_0a8ecf"
        id="selectListbox"
        role="listbox"
      >
        <li
          aria-selected="false"
          class="_root_b75fe4"
          data-option-value="32"
          role="option"
          tabindex="0"
        >
          My Game 1
        </li>
        <li
          aria-selected="false"
          class="_root_b75fe4"
          data-option-value="51"
          role="option"
          tabindex="0"
        >
          My Game 2
        </li>
        <li
          aria-selected="false"
          class="_root_b75fe4"
          data-option-value="77"
          role="option"
          tabindex="0"
        >
          Game with a really real...
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
